packages/flutter_tools/lib/src/macos/xcode.dart

```diff
@@ -185,3 +190,274 @@ class Xcode {
     );
   }
 }
+
+/// A utility class for interacting with Xcode xcdevice command line tools.
+class XCDevice {
+  XCDevice({
+    @required ProcessManager processManager,
+    @required Logger logger,
+    @required Xcode xcode,
+  }) : _processUtils = ProcessUtils(logger: logger, processManager: processManager),
+       _logger = logger,
+       _xcode = xcode;
+
+  final ProcessUtils _processUtils;
+  final Logger _logger;
+  final Xcode _xcode;
+
+  bool get isInstalled => _xcode.isInstalledAndMeetsVersionCheck && xcdevicePath != null;
+
+  String _xcdevicePath;
+  String get xcdevicePath {
+    if (_xcdevicePath == null) {
+      try {
+        _xcdevicePath = _processUtils.runSync(
+          <String>[
+            'xcrun',
+            '--find',
+            'xcdevice'
+          ],
+        ).stdout.trim();
+      } on ProcessException {
+        // Ignored, return null below.
+      } on ArgumentError {
+        // Ignored, return null below.
+      }
+    }
+    return _xcdevicePath;
+  }
+
+  Future<List<dynamic>> _getAllDevices() async {
+    if (!isInstalled) {
+      return null;
+    }
+    try {
+      // USB-tethered devices should always be found.
+      // Finding networked devices takes longer than the default timeout. Try a 5-second timeout.
+      final RunResult result = await _processUtils.run(
+        <String>[
+          xcdevicePath,
+          'list',
+          '--timeout',
```

Comment:
Can we pass a smaller timeout if we know the user doesn't care about wireless debugging?

Said another way, if wireless device discovery was under a flag, could this number be smaller?

Reply:
If the user doesn't care about networked devices and this is changed to `--timeout 1` (faster than the default) , this returns the USB-tethered devices and none of the networked devices.  

There's an `xcdevice wait [--usb|--wifi|--both] [--timeout=sec] <udid>` which will return as soon as the device is found but waits until the timeout if not, which would be fast for USB-tethered devices.  That would make sense for `flutter run -d <udid>` `hasSpecifiedDeviceId` situations, though it would require `PollingDeviceDiscovery` to get refactored to take a "hint" identifier or something like that.

Maybe it would be okay for `flutter doctor` and `flutter devices` to take the full 5 seconds since we can then surface that it's networked and that they need to pass in `-d <udid>` to  `flutter run` (`attach`, `logs`, `screenshot`, etc) on the device.  And users would only be running those command infrequently, and they probably would prefer an accurate listing than for them to be fast.

Proposal:
1. Switch to `xcdevice wait --both --timeout=5` for any command using `-d <udid>` (`run`, `attach`, `logs`, `screenshot`) to return fast if possible, but wait if needed.
1. When there's no `-d <udid>` or `-d all` assume USB-only (common case) and use the fastest timeout possible.
1. Keep the long timeout for `flutter doctor` and `flutter devices` and for "network" interface show a warning like "This device is available over the network and must be specified with `-d {}

Reply:
Can the `--timeout` value be smaller than 1?

Reply:
> Can the `--timeout` value be smaller than 1?

1 seems to be the minimum allowed.  0 reverts to the longer default timeout.